src/backend/cpu/nearest_neighbour.cpp

```diff
@@ -26,35 +27,38 @@ void nearest_neighbour(Array<uint>& idx, Array<To>& dist,
                        const uint dist_dim, const uint n_dist,
                        const af_match_type dist_type)
 {
-    if (n_dist > 1) {
-        CPU_NOT_SUPPORTED("\nNumber of smallest distances can't be <1\n");
-    }
```

Comment:
This check should be present atleast in CUDA backend where there is a limitation of `n_dist`, 256 because of topk.

Reply:
Added as an ARG_ASSERT to api/c/ since change affects all backends